File: Editorial/3601-3700/3633_Earliest_Finish_Time_for_Land_and_Water_Rides_I/CSharp/1_Brute_Force_Enumeration_Classification_Discussion.cs
public class Solution {
    public int EarliestFinishTime(int[] landStartTime, int[] landDuration,
                                  int[] waterStartTime, int[] waterDuration) {
        int n = landStartTime.Length;
        int m = waterStartTime.Length;
        int res = int.MaxValue;
        ;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                int land = landStartTime[i] + landDuration[i];
                int land_water =
                    Math.Max(land, waterStartTime[j]) + waterDuration[j];
                res = Math.Min(res, land_water);

                int water = waterStartTime[j] + waterDuration[j];
                int water_land =
                    Math.Max(water, landStartTime[i]) + landDuration[i];
                res = Math.Min(res, water_land);
            }
        }
        return res;
    }
}

File: Editorial/3601-3700/3633_Earliest_Finish_Time_for_Land_and_Water_Rides_I/TypeScript/1_Brute_Force_Enumeration_Classification_Discussion.ts
function earliestFinishTime(
    landStartTime: number[],
    landDuration: number[],
    waterStartTime: number[],
    waterDuration: number[],
): number {
    let n = landStartTime.length;
    let m = waterStartTime.length;
    let res = Infinity;
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < m; j++) {
            let land = landStartTime[i] + landDuration[i];
            let land_water =
                Math.max(land, waterStartTime[j]) + waterDuration[j];
            res = Math.min(res, land_water);

            let water = waterStartTime[j] + waterDuration[j];
            let water_land =
                Math.max(water, landStartTime[i]) + landDuration[i];
            res = Math.min(res, water_land);
        }
    }
    return res;
}
